parlai/core/torch_generator_agent.py

```diff
@@ -1512,9 +1548,11 @@ def get_rescored_finished(self, n_best=None):
             number of finalized hypotheses to return
 
         :return:
-            list of (tokens, score) pairs, in sorted order, where:
+            list of (tokens, score, token_scores, token_ranks) pairs, in sorted order, where:
```

Comment:
Is it worth us refactoring this away from a tuple and to some sort of Named class at this point?

Reply:
Probably, yes. Before we do that though, we should probably add type hints to all of the methods?

Reply:
That could make things less error prone definitely

Reply:
There are only two callsites for this method, and they are in the same method. I think it is okay to leave as is for now.